src/constants/actions.js

```diff
@@ -0,0 +1,19 @@
+module.exports = {
+  // action types
+  TRANSITION: 'ACTION_TYPE/TRANSITION',
+  METADATA: 'ACTION_TYPE/METADATA',
+  PROGRESS: 'ACTION_TYPE/PROGRESS',
+  // transition actions
+  REJECT: 'TRANSITION/REJECT',
```

Comment:
I view `state_transition` to be a virtual sub-resource of a pin. It can be created only. And, it depends on previous state. So, what we designed would be fine.

The thing that annoys me is `metadata_change`.  Actually, I prefer just doing CRUD on `/pins/{pinId}`.  However, currently, we use `feathers-mongoose`. We have to get previous metadata before the mongoose service update the values and have to write log by using that previous metadata.